lib/workers/package-file/index.js

```diff
@@ -96,6 +96,7 @@ async function renovatePackageFile(packageFileConfig) {
     'devDependencies',
     'optionalDependencies',
     'peerDependencies',
+    'engines',
```

Comment:
`engines` seem sufficiently different from dependencies that it should be it's own top-level feature for npm (As in, it shouldn't be forced into the same upgrade workflow as normal package dependencies).

Reply:
I think it will make logic a lot easier if we can keep the linking of "only one package manager per file", i.e. `package.json` belongs to the `npm` "manager" in Renovate.

The funny thing to me was that I didn't need to touch 50% of the functionality to get this done, e.g. extracting or updating/replacing versions in the file - so it's a better fit than you or I perhaps expected. Also, the entire concept of engines was created for `npm` so it seems reasonable that our npm manager takes care of them. It might be different if we consider the cases where other apps add their config to `package.json` (e.g. like Renovate supports) however that's not the case here.

>  it shouldn't be forced into the same upgrade workflow

I don't see how anything is "forced" or undesirable but I may be missing something.

Reply:
> I don't see how anything is "forced" or undesirable but I may be missing something.

I think the example below, wherechangelog fetching is short circuited for the `node` engine property, is a good example of where the existing dependency upgrade pipeline doesn't quite apply the same to an `engines` upgrade.

The short circuit is a minor item. I just worry that those types of exceptions could grow further as additional enhancements are made to the `engines` feature.